### DISCUSSION TITLE
[@types/lodash] Issue with lodash typings causing TypeScript compilation error

### DISCUSSION THREAD

**Discussion by nikbarseghyan** [DefinitelyTyped/DefinitelyTyped#65742] [Issues with a @types package] [answered]
Hi all
I'm encountering an error related to the Lodash typings while trying to compile my TypeScript project. The issue seems to be with the compatibility between the Lodash typings and TypeScript. I have followed several troubleshooting steps, including updating TypeScript, downgrading Lodash typings, and reviewing my project dependencies, but the error persists.

Environment:
TypeScript version:  v5.1.3
Node.js version: v20

The TypeScript compilation fails with the following error message:
`yarn run build
yarn run v1.22.19
$ tsc
node_modules/@types/lodash/common/common.d.ts:194:15 - error TS2589: Type instantiation is excessively deep and possibly infinite.

194     interface Object<T> extends LoDashImplicitWrapper<T> {
                  ~~

node_modules/@types/lodash/common/common.d.ts:206:15 - error TS2430: Interface 'ObjectChain<T>' incorrectly extends interface 'LoDashExplicitWrapper<T>'.
  The types returned by 'entries().pop()' are incompatible between these types.
    Type 'CollectionChain<string | T[keyof T]>' is missing the following properties from type 'ObjectChain<[string, any]>': assign, assignIn, assignInWith, assignWith, and 11 more.

206     interface ObjectChain<T> extends LoDashExplicitWrapper<T> {
                  ~~~~~

Found 2 errors in the same file, starting at: node_modules/@types/lodash/common/common.d.ts:194`

The problem is that I don't use @types/lodash library in my project

tsconfig
`
{
	"compilerOptions": {
           ...
	},
	"exclude": [
		"node_modules",
		"node_modules/@types/lodash",
		"./dist/*/",
	]
}
`

I would greatly appreciate any help or guidance to resolve this issue. If you have any suggestions, workarounds, or insights into the compatibility between Lodash typings and TypeScript, please let me know. Thank you!

[a comment by an automated account is not shown]

**Comment by bschlenk** [comment, marked as answer]
I upgraded to typescript 5.1.6 and the issue went away, so it may have been a typescript bug.

**Reply by nikbarseghyan** [reply]
ok thanks
